### PR TITLE
Use async client to delete scroll and pit for OpenSearch as workaroun…

### PR DESCRIPTION
**Pull request by graytaylor0**
…d for bug in client

### Description
The OpenSearch Java Client has an issue where delete requests with sigv4 are not signed correctly (https://github.com/opensearch-project/opensearch-java/issues/521). 

As a workaround, this change constructs an async OpenSearch client to make only the delete calls for scroll and pit
 
### Check List
- [x] New functionality includes testing.
- [ ] New functionality has a documentation issue. Please link to it in this PR.
  - [ ] New functionality has javadoc added
- [x] Commits are signed with a real name per the DCO

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: data-prepper-plugins/opensearch-source/src/main/java/org/opensearch/dataprepper/plugins/source/opensearch/worker/client/OpenSearchAccessor.java [hunk 1]
```diff
@@ -126,7 +130,7 @@ public SearchWithSearchAfterResults searchWithPit(final SearchPointInTimeRequest
     @Override
     public void deletePit(final DeletePointInTimeRequest deletePointInTimeRequest) {
         try {
-            final DeletePitResponse deletePitResponse = openSearchClient.deletePit(DeletePitRequest.of(builder -> builder.pitId(Collections.singletonList(deletePointInTimeRequest.getPitId()))));
+            final DeletePitResponse deletePitResponse = openSearchAsyncClient.deletePit(DeletePitRequest.of(builder -> builder.pitId(Collections.singletonList(deletePointInTimeRequest.getPitId()))));
```
Comment (dlvenable) [inline data-prepper-plugins/opensearch-source/src/main/java/org/opensearch/dataprepper/plugins/source/opensearch/worker/client/OpenSearchAccessor.java:133]: So this is still using the existing synchronous method, but the implementation is the "async" client. Right? Why not use only the async client (but in the synchronous way)?

**Comment by dlvenable** [comment]
Also, [this comment](https://github.com/opensearch-project/opensearch-java/issues/521#issuecomment-1585322675) indicates that the problem may be the Apache HTTP client.

Can we skip using the async client and just change to use Netty underneath?

File: data-prepper-plugins/opensearch-source/src/main/java/org/opensearch/dataprepper/plugins/source/opensearch/worker/client/OpenSearchAccessor.java (hunk 1 above)
Comment (graytaylor0) [reply data-prepper-plugins/opensearch-source/src/main/java/org/opensearch/dataprepper/plugins/source/opensearch/worker/client/OpenSearchAccessor.java:133]: The only netty client is the `NettyNioAsyncHttpClient`. Are you suggesting using this one?

**Comment by graytaylor0** [comment]
> Also, [this comment](https://github.com/opensearch-project/opensearch-java/issues/521#issuecomment-1585322675) indicates that the problem may be the Apache HTTP client.
> 
> Can we skip using the async client and just change to use Netty underneath?

The only netty client is the NettyNioAsyncHttpClient. Are you suggesting using this one?

**Comment by dlvenable** [comment]
> The only netty client is the NettyNioAsyncHttpClient. Are you suggesting using this one?

Yes.

As I'm reading this, we have two clients - sync and async. We use them both for synchronous APIs. So why not just change the existing sync client to use `NettyNioAsyncHttpClient` instead of  `ApacheHttpClient`? Then we could have a single client used everywhere.

**Review by dlvenable — APPROVED** [review]
Nice! I'm glad to see that simplification works.

**Comment by graytaylor0** [comment]
> Nice! I'm glad to see that simplification works.

Me as well! Good call

**Review by travisbenedict — APPROVED** [review]

**Pull request merged**